pkg/virt-handler/vm.go

```diff
@@ -4,43 +4,48 @@ import (
 	"fmt"
 	kubeapi "k8s.io/client-go/1.5/pkg/api"
 	kubev1 "k8s.io/client-go/1.5/pkg/api/v1"
+	"k8s.io/client-go/1.5/rest"
 	"k8s.io/client-go/1.5/tools/cache"
 	"k8s.io/client-go/1.5/tools/record"
 	"kubevirt.io/kubevirt/pkg/api/v1"
 	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/logging"
 	"kubevirt.io/kubevirt/pkg/virt-handler/libvirt"
 )
 
-func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.DomainManager, recorder record.EventRecorder) (cache.Indexer, *cache.Controller) {
+func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.DomainManager, recorder record.EventRecorder, restClient rest.RESTClient) (cache.Indexer, *cache.Controller) {
+	logger := logging.DefaultLogger()
+
 	return kubecli.NewInformer(listWatcher, &v1.VM{}, 0, kubecli.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) error {
-			fmt.Printf("VM ADD\n")
+			logger.Info().Log("VM ADD")
 			vm := obj.(*v1.VM)
 			err := domainManager.SyncVM(vm)
 			if err != nil {
-				fmt.Println(err)
-				recorder.Event(vm, kubev1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
-				return cache.ErrRequeue{Err: err}
+				goto is_error
 			}
 
-			restClient, err := kubecli.GetRESTClient()
-			if err != nil {
-				return cache.ErrRequeue{Err: err}
+			if vm.Status.Phase != v1.Running {
+				vm.Status.Phase = v1.Running
+				err = restClient.Put().Resource("vms").Body(vm).
+					Name(vm.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error()
+				if err != nil {
+					goto is_error
+				}
 			}
-			vm.Status.Phase = v1.Succeeded
-			err = restClient.Put().Resource("vms").Body(vm).
-				Name(vm.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error()
+		is_error:
 			if err != nil {
+				logger.Error().Log(err)
```

Comment:
The logger is a structured logger. It can be used like:

logger.Error().Log("msg", err)

or

logger.Error().Msg(err)